internal/server/proto/server.proto

```diff
@@ -2828,6 +2831,25 @@ message GetStatusReportRequest {
   // a report by deployment or release operation id
 }
 
+message ExpediteStatusReportRequest {
```

Comment:
Should this be one of them newfangled `UI_` endpoints instead?

It feels weird to have a special API for queueing a status report when we have the generic `QueueJob` that supports this (and internally we're just building that up anyways).

Reply:
That's a good point.... @jgwhite should the UI just be using the `QueueJob` API endpoint? This service func isn't doing anything that the frontend could handle if that's the case.

Reply:
That’s a good question. Originally we were planning to call `QueueJob` directly from the UI but as we discussed it we realized that we wanted something that would queue the job whilst respecting the new app polling logic (rather than sidestepping it). To me this doesn’t feel like it’s explicitly tailored to the UI but I’m certainly not opposed to adding the `UI_` prefix.

Reply:
Ah right, I think we wouldn't have to worry about the app status polling logic, since this sidesteps that as well and queues a job directly rather than trying to update an apps "next poll" to now. So maybe instead of this the front end queues the job?

Reply:
On the other hand, it could be worth having a dedicated endpoint that does all of the construction to refresh a status report 🤔

Reply:
> So maybe instead of this the front end queues the job?

Personally I’d like the UI to use `QueueJob` directly. The one argument I can see against it is the special logic around constructing the singleton ID:

https://github.com/hashicorp/waypoint/blob/d0fdb750fc8b79d9fdd39c73e54750a607d06f09/internal/server/singleprocess/service_status_report.go#L127-L129

We could certainly recreate this in the UI but it might be a little brittle? Maybe I’m being over-cautious?

Reply:
You should be able to use `SingletonId` when constructing a job from the UI and sending it through the `QueueJob` endpoint. I think it's nice in this case too prevent people from accidentally clicking refresh and queuing a bunch of jobs?

Reply:
Chiming in here!
@jgwhite and I just tried walking through a `QueueJob` implementation and there's ...a lot... of code to send a request. An issue is that `StatusReportOp` needs a `Deployment` or a `Release`, not a `Ref.Deployment` or `Ref.Release`. We have all the information about the deployment or release available, but in object format. The JS protobuf library does _**not**_ have the ability to convert an object into a specified class.  Here's just some of the code:
```
    let app = new Ref.Application();
    app.setApplication(...);
    app.setProject(...);

    let workspace = new Ref.Workspace();
    workspace.setWorkspace(...)

    let runner = new Ref.Runner();
    runner.setAny(new Ref.RunnerAny());

    let statusReportOp = new Job.StatusReportOp();
    // NOTE: This is where we would have to create a new Deployment()
    // which adds even more lines to this than what is already here
    statusReportOp.setDeployment();

    let job = new Job();
    job.setApplication(app);
    job.setWorkspace(workspace);
    job.setTargetRunner(runner);
    job.setStatusReport(statusReportOp);

    let req = new QueueJobRequest();
    req.setJob(job);
    let resp = await this.api.client.queueJob(req, this.api.WithMeta());
```
This might be an argument to have a separate `ExpediteStatusReport` function, which we can add the `UI_` prefix to

Reply:
One other option, which we do for HCL formatting for example, is have a `UI_` endpoint that just returns the pre-populated `QueueJobRequest` that it would’ve sent. That way its more flexible in the future if the UI ever wants to tweak it. So could do like `UI_StatusReportQueueJobRequest` or something (prob too long) and just get the right structure back.

I’m not against the original approach either if that makes sense, just trying to give you the most flexible tools.

Reply:
Just to circle back to this thread....we discussed this in slack and we are thinking for now this is OK as is. In the future if the UI ever needs a more tweakable status report refresh job, we should introduce an endpoint that returns a `QueuedJobRequest` proto message instead and let the UI manipulate the request before queuing.